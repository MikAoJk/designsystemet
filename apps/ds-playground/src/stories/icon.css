.myIconClassname {
  fill: var(--palette-denim-100);
}
