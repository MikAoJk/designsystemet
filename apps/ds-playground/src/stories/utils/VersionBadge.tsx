import { ReactNode } from 'react';

interface VersionBadgeProps {
  packageName: string;
}

export const VersionBadge = ({ packageName }: VersionBadgeProps): ReactNode => {
  let color = 'green';

  let version = process.env[`@skatteetaten/ds-${packageName}`] ?? '..';
  if (process.env['NODE_ENV'] === 'development') {
    color = 'red';
    version = 'developing';
  }
  if (process.env['STORYBOOK_BUILD_VERSION']?.includes('SNAPSHOT')) {
    color = 'blue';
  }

  const normalizedPackageName = packageName.replaceAll('-', '--');
  const normalizedVersion = version.replaceAll('-', '--').replaceAll('_', '__');

  return (
    <img
      src={`https://img.shields.io/badge/%40skatteetaten%2Fds--${normalizedPackageName}-${normalizedVersion}-${color}?logo=npm`}
      alt={`@skatteetaten/ds-${packageName}@${version}`}
    />
  );
};
