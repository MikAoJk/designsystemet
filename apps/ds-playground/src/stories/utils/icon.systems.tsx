import { SystemSVGPathsAndIcons } from '@skatteetaten/ds-icons';

import {
  findSVGPathElements,
  makeLabelsFromSVGPathElements,
} from './icon.utils';

export const SystemSVGPaths = findSVGPathElements(SystemSVGPathsAndIcons);
export const SystemIconLabels = makeLabelsFromSVGPathElements(SystemSVGPaths);
