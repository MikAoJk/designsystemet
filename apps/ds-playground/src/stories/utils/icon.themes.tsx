import { ThemeSVGPathsAndIcons } from '@skatteetaten/ds-icons';

import {
  findSVGPathElements,
  makeLabelsFromSVGPathElements,
} from './icon.utils';

export const ThemeSVGPaths = findSVGPathElements(ThemeSVGPathsAndIcons);
export const ThemeIconLabels = makeLabelsFromSVGPathElements(ThemeSVGPaths);
