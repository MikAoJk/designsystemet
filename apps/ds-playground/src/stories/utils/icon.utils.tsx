import {
  SystemSVGPathsAndIcons,
  ThemeSVGPathsAndIcons,
} from '@skatteetaten/ds-icons';

export type SVGPathAndIconType =
  | typeof SystemSVGPathsAndIcons
  | typeof ThemeSVGPathsAndIcons;
export type SVGPathAndIconKey = keyof SVGPathAndIconType;

export const makeLabelsFromSVGPathElements = function (
  SVGPathElements: Record<SVGPathAndIconKey, SVGPathAndIconType>
): Record<string, string> {
  const labels: Record<string, string> = {};
  Object.keys(SVGPathElements).forEach((name: string) => {
    labels[name] = name.replace('SVGpath', '');
  });

  return labels;
};

export const findSVGPathElements = function (
  SVGPathAndIconElements: Record<SVGPathAndIconKey, SVGPathAndIconType>
): Record<string, SVGPathAndIconType[SVGPathAndIconKey]> {
  const elements: Record<string, SVGPathAndIconType[SVGPathAndIconKey]> = {};
  Object.keys(SVGPathAndIconElements).forEach((elementName: string) => {
    if (elementName.endsWith('SVGpath')) {
      elements[elementName] =
        SVGPathAndIconElements[elementName as SVGPathAndIconKey];
    }
  });

  return elements;
};
