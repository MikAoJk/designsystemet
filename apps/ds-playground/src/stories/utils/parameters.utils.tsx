import { ArgsTable, Stories, PRIMARY_STORY } from '@storybook/addon-docs';
import type { Parameters } from '@storybook/csf';

type DesignParameters = {
  name: string;
  url: string;
};

export const getParameters = function (
  designParameters: DesignParameters[]
): Parameters {
  return {
    docs: {
      page: (): JSX.Element => {
        return (
          <>
            <Stories title={''} />
            <ArgsTable story={PRIMARY_STORY} />
          </>
        );
      },
    },
    design: designParameters.map((designParameter: DesignParameters) => {
      return {
        name: designParameter.name,
        type: 'figma',
        url: designParameter.url,
      };
    }),
    backgrounds: {
      values: [
        { name: 'Svart', value: '#000' },
        { name: 'Hvit', value: '#fff' },
      ],
    },
  };
};
