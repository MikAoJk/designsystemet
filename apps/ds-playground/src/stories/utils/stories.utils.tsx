export const exampleParameters = {
  docs: {
    source: {
      type: 'code',
      language: 'tsx',
      transform: (source: string): string => {
        return source
          .replace('render: (_args): JSX.Element', 'const Examples = ()')
          .replace(
            'render: function Render(_args): JSX.Element ',
            'const Examples = () => '
          )
          .replace('satisfies Story', '');
      },
    },
  },
  /* _args i en story må finnes slik at props i fanen Controls skjules og er prefikset med underscore for å unngå eslint klaging */
  controls: {
    exclude: /.*/,
  },
};
