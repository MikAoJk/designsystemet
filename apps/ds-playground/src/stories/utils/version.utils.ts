export const getVersion = (
  packageName: string
): {
  major: string;
  minor: string;
  patch: string;
  postfix?: string;
  style?: Record<string, string>;
} => {
  const ver = process.env[`@skatteetaten/${packageName}`] ?? '..';
  const [major, minor, patch] = ver.split('.');
  const majorWithPackageName = packageName ? `${packageName}@${major}` : major;

  const style = {
    color: 'var(--semantic-page-foreground)',
    background: 'var(--semantic-success-background)',
    'font-weight': 'var(--font-weight-bold)',
    'font-size': '13px',
  };

  if (process.env['NODE_ENV'] === 'development') {
    return {
      major: majorWithPackageName,
      minor,
      patch,
      postfix: 'develop',
      style: {
        ...style,
        background: 'var(--semantic-danger-background)',
      },
    };
  }
  if (process.env['STORYBOOK_BUILD_VERSION']?.includes('SNAPSHOT')) {
    return {
      major: majorWithPackageName,
      minor,
      patch,
      postfix: process.env['STORYBOOK_BUILD_VERSION'],
      style: {
        ...style,
        background: 'var(--semantic-warning-background)',
      },
    };
  }

  return {
    major: majorWithPackageName,
    minor,
    patch,

    style: {
      ...style,
    },
  };
};
