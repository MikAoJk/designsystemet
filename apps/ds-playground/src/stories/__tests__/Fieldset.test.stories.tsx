import { Fieldset, FieldsetProps } from '@skatteetaten/ds-forms';
import { WarningSVGpath } from '@skatteetaten/ds-icons';
import { Heading, Paragraph } from '@skatteetaten/ds-typography';
import { expect } from '@storybook/jest';
import { Meta, StoryObj } from '@storybook/react';
import { fireEvent, within } from '@storybook/testing-library';

import { wrapper } from './testUtils/storybook.testing.utils';
import { SystemSVGPaths } from '../utils/icon.systems';

const meta = {
  component: Fieldset,
  title: 'Tester/Fieldset',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: { table: { disable: true } },
    description: { table: { disable: true } },
    hasSpacing: { table: { disable: true } },
    helpText: { table: { disable: true } },
    helpSvgPath: {
      table: { disable: true },
      options: Object.keys(SystemSVGPaths),
      mapping: SystemSVGPaths,
    },
    hideLegend: { table: { disable: true } },
    legend: { table: { disable: true } },
    showRequiredMark: { table: { disable: true } },
    titleHelpSvg: { table: { disable: true } },
    // HTML
    disabled: { table: { disable: true } },
  },
} satisfies Meta<typeof Fieldset>;
export default meta;
type Story = StoryObj<typeof meta>;

const defaultDescription = 'Kort hjelpetekst';
const defaultHelpText =
  'Vi trenger å vite navnet ditt dersom vi skal kontakte deg senere.';
const defaultLegendText = 'Ledetekst';
const defaultArgs: FieldsetProps = {
  legend: defaultLegendText,
  children: <div>{'Innhold'}</div>,
};

export const WithRef = {
  name: 'With Ref (FA1)',
  args: {
    ...defaultArgs,
    ref: (instance: HTMLFieldSetElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },
  argTypes: {
    ref: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const fieldset = canvas.getByRole('group');
    await expect(fieldset).toHaveAttribute('id', 'dummyIdForwardedFromRef');
  },
} satisfies Story;

export const WithAttributes = {
  name: 'With Attributes (FA2-5)',
  args: {
    ...defaultArgs,
    id: 'htmlid',
    className: 'dummyClassname',
    lang: 'nb',
    'data-testid': '123ID',
  },
  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const fieldset = canvas.getByRole('group');
    await expect(fieldset).toHaveAttribute('id', 'htmlid');
    await expect(fieldset).toHaveClass('dummyClassname');
    await expect(fieldset).toHaveAttribute('lang', 'nb');
    await expect(fieldset).toHaveAttribute('data-testid', '123ID');
  },
} satisfies Story;

export const Defaults = {
  name: 'Default (FS-A1, FS-A2, FS-A4, A2, A3, B1)',
  args: {
    ...defaultArgs,
  },
  argTypes: {
    legend: { table: { disable: false } },
    children: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const fieldset = canvas.getByRole('group');
    await expect(fieldset.tagName).toBe('FIELDSET');
    // eslint-disable-next-line testing-library/no-node-access
    await expect(fieldset.firstElementChild?.tagName).toBe('LEGEND');
  },
} satisfies Story;

export const WithDescription = {
  name: 'With Description (FS-A3)',
  args: {
    ...defaultArgs,
    description: defaultDescription,
  },
  argTypes: {
    description: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const descriptionNode = canvas.getByText(defaultDescription);
    await expect(descriptionNode).toBeInTheDocument();
  },
} satisfies Story;

export const WithShowRequiredMark = {
  name: 'With ShowRequiredMark (FS-A4)',
  args: {
    ...defaultArgs,
    showRequiredMark: true,
  },
  argTypes: {
    showRequiredMark: { table: { disable: false } },
  },
} satisfies Story;

export const WithShowRequiredMarkAndLegend = {
  name: 'With ShowRequiredMark And Legend Contains Markup (FS-A4)',
  args: {
    ...defaultArgs,
    legend: (
      <>
        <Heading as={'h1'} level={3}>
          {'Dette er en Heading i legend'}
        </Heading>
        <Paragraph variant={'ingress'}>
          <em>{'Dette er en italic Paragraph med ingress variant i legend'}</em>
        </Paragraph>
      </>
    ),
    showRequiredMark: true,
  },
  argTypes: {
    legend: {
      table: { disable: true },
      control: { type: null },
    },
    showRequiredMark: { table: { disable: false } },
  },
} satisfies Story;

export const WithHideLegend = {
  name: 'With HideLegend (FS-A7)',
  args: {
    ...defaultArgs,
    description: defaultDescription,
    helpText: defaultHelpText,
    hideLegend: true,
  },
  argTypes: {
    hideLegend: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const legend = canvas.getByText(defaultLegendText);
    const descriptionNode = canvas.getByText(defaultDescription);
    const helpButtonNode = canvas.getByRole('button');
    await expect(legend).toBeInTheDocument();
    await expect(descriptionNode).toBeInTheDocument();
    await expect(helpButtonNode).toBeInTheDocument();
  },
} satisfies Story;

export const WithSpacing = {
  name: 'With Spacing (A2)',
  args: {
    ...defaultArgs,
    hasSpacing: true,
  },
  argTypes: {
    hasSpacing: { table: { disable: false } },
  },
} satisfies Story;

export const WithDisabled = {
  name: 'With Disabled (A4)',
  args: {
    ...defaultArgs,
    disabled: true,
  },
  argTypes: {
    disabled: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const fieldset = canvas.getByRole('group');
    await expect(fieldset).toBeDisabled();
  },
} satisfies Story;

export const WithHelpText = {
  name: 'With HelpText (FS-A1, FS-A5 delvis, FS-B2 delvis)',
  args: {
    ...defaultArgs,
    helpText: defaultHelpText,
  },
  argTypes: {
    helpText: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: {
      focus: `${wrapper} > fieldset > button`,
      click: `${wrapper} > fieldset > button`,
    },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const helpButton = canvas.getByRole('button', {
      description: defaultLegendText,
    });
    await expect(helpButton).toBeInTheDocument();
    await expect(helpButton).toHaveAttribute('aria-expanded', 'false');
    const legend = canvas.getByText(defaultLegendText);
    await expect(helpButton).toHaveAttribute('aria-describedby', legend.id);
    const helpSvg = canvas.getByLabelText('Hjelp', { selector: 'svg' });
    await expect(helpSvg).toBeInTheDocument();
  },
} satisfies Story;

export const WithHelpTextAndDescription = {
  name: 'With HelpText And Description (A1)',
  args: {
    ...defaultArgs,
    helpText:
      'Vi trenger å vite navnet ditt dersom vi skal kontakte deg senere.',
    description: 'En liten beskrivelse tekst',
  },
  argTypes: {
    helpText: { table: { disable: false } },
    description: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: {
      focus: `${wrapper} > fieldset > button`,
      click: `${wrapper} > fieldset > button`,
    },
  },
} satisfies Story;

export const WithHelpSvgPath = {
  name: 'With HelpSvgPath (FS-A5 delvis)',
  args: {
    ...defaultArgs,
    helpText: defaultHelpText,
    helpSvgPath: WarningSVGpath,
  },
  argTypes: {
    helpSvgPath: {
      table: { disable: false },
    },
  },
} satisfies Story;

export const ClickHelpButton = {
  name: 'Click HelpButton On And Off (FS-A5 delvis, FS-B2 delvis)',
  args: {
    ...defaultArgs,
    helpText: defaultHelpText,
  },
  parameters: {
    imageSnapshot: {
      disable: true,
    },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const helpButton = canvas.getByRole('button', {
      description: defaultLegendText,
    });
    await expect(helpButton).toHaveAttribute('aria-expanded', 'false');
    await fireEvent.click(helpButton);
    await expect(helpButton).toHaveAttribute('aria-expanded', 'true');
    const helpText = canvas.getByText(defaultHelpText);
    await expect(helpText).toBeInTheDocument();
    await fireEvent.click(helpButton);
    await expect(helpText).not.toBeInTheDocument();
    await expect(helpButton).toHaveAttribute('aria-expanded', 'false');
  },
} satisfies Story;

export const ClickCloseButton = {
  name: 'Close HelpText With CloseButton (FS-A5 delvis, FS-A6, FS-B2 delvis)',
  args: {
    ...defaultArgs,
    helpText: defaultHelpText,
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const helpButton = canvas.getByRole('button', {
      description: defaultLegendText,
    });
    await fireEvent.click(helpButton);
    const closeButton = canvas.getByTitle('Lukk');
    await expect(closeButton).toBeInTheDocument();
    const helpText = canvas.getByText(defaultHelpText);
    await fireEvent.click(closeButton);
    await expect(helpText).not.toBeInTheDocument();
    await expect(helpButton).toHaveFocus();
  },
} satisfies Story;
