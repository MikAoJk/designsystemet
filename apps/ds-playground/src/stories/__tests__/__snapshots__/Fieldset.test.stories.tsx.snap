// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/Fieldset ClickCloseButton play-test 1`] = `
<div data-test-block="true">
  <fieldset class="Fieldset_fieldset__gwnkr">
    <legend id="legendId-:r1a:"
            class="Fieldset_legend__8odl-   Fieldset_legendNoMarginBottom__UO3iu"
    >
      Ledetekst
    </legend>
    <button class="IconButton_button__mw9sG IconButton_button_extraSmall__3+QfY IconButton_button_outlined__aDbNH Fieldset_helpButton__Qd9mY "
            type="button"
            aria-describedby="legendId-:r1a:"
            aria-expanded="false"
    >
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz IconButton_icon__1J4sN"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:r1b:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:r1b:">
          Hjelp
        </title>
        <path d="M10 19h3v3h-3v-3Zm2-17c5.35.22 7.68 5.62 4.5 9.67-.83 1-2.17 1.66-2.83 2.5C13 15 13 16 13 17h-3c0-1.67 0-3.08.67-4.08.66-1 2-1.59 2.83-2.25C15.92 8.43 15.32 5.26 12 5a3 3 0 0 0-3 3H6a6 6 0 0 1 6-6Z">
        </path>
      </svg>
    </button>
    <div class="Fieldset_contentContainer__fJcfT ">
      <div class>
        <label for="textFieldTextboxId-:r1d:"
               class="TextField_label__K0isy  undefined"
        >
          <span class>
            Label
          </span>
        </label>
        <input id="textFieldTextboxId-:r1d:"
               class="TextField_textbox__qbZlk    undefined"
               value
        >
        <div id="textFieldErrorId-:r1c:"
             aria-live="assertive"
             aria-atomic="true"
        >
        </div>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Tester/Fieldset ClickHelpButton play-test 1`] = `
<div data-test-block="true">
  <fieldset class="Fieldset_fieldset__gwnkr">
    <legend id="legendId-:r15:"
            class="Fieldset_legend__8odl-   Fieldset_legendNoMarginBottom__UO3iu"
    >
      Ledetekst
    </legend>
    <button class="IconButton_button__mw9sG IconButton_button_extraSmall__3+QfY IconButton_button_outlined__aDbNH Fieldset_helpButton__Qd9mY "
            type="button"
            aria-describedby="legendId-:r15:"
            aria-expanded="false"
    >
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz IconButton_icon__1J4sN"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:r16:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:r16:">
          Hjelp
        </title>
        <path d="M10 19h3v3h-3v-3Zm2-17c5.35.22 7.68 5.62 4.5 9.67-.83 1-2.17 1.66-2.83 2.5C13 15 13 16 13 17h-3c0-1.67 0-3.08.67-4.08.66-1 2-1.59 2.83-2.25C15.92 8.43 15.32 5.26 12 5a3 3 0 0 0-3 3H6a6 6 0 0 1 6-6Z">
        </path>
      </svg>
    </button>
    <div class="Fieldset_contentContainer__fJcfT ">
      <div class>
        <label for="textFieldTextboxId-:r18:"
               class="TextField_label__K0isy  undefined"
        >
          <span class>
            Label
          </span>
        </label>
        <input id="textFieldTextboxId-:r18:"
               class="TextField_textbox__qbZlk    undefined"
               value
        >
        <div id="textFieldErrorId-:r17:"
             aria-live="assertive"
             aria-atomic="true"
        >
        </div>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Tester/Fieldset Defaults play-test 1`] = `
<div data-test-block="true">
  <fieldset class="Fieldset_fieldset__gwnkr">
    <legend id="legendId-:r6:"
            class="Fieldset_legend__8odl-"
    >
      Ledetekst
    </legend>
    <div class="Fieldset_contentContainer__fJcfT ">
      <div class>
        <label for="textFieldTextboxId-:r8:"
               class="TextField_label__K0isy  undefined"
        >
          <span class>
            Label
          </span>
        </label>
        <input id="textFieldTextboxId-:r8:"
               class="TextField_textbox__qbZlk    undefined"
               value
        >
        <div id="textFieldErrorId-:r7:"
             aria-live="assertive"
             aria-atomic="true"
        >
        </div>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Tester/Fieldset WithAttributes play-test 1`] = `
<div data-test-block="true">
  <fieldset id="htmlid"
            data-testid="123ID"
            class="Fieldset_fieldset__gwnkr  dummyClassname"
            lang="nb"
  >
    <legend id="legendId-:r3:"
            class="Fieldset_legend__8odl-"
    >
      Ledetekst
    </legend>
    <div class="Fieldset_contentContainer__fJcfT ">
      <div class>
        <label for="textFieldTextboxId-:r5:"
               class="TextField_label__K0isy  undefined"
        >
          <span class>
            Label
          </span>
        </label>
        <input id="textFieldTextboxId-:r5:"
               class="TextField_textbox__qbZlk    undefined"
               value
        >
        <div id="textFieldErrorId-:r4:"
             aria-live="assertive"
             aria-atomic="true"
        >
        </div>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Tester/Fieldset WithDescription play-test 1`] = `
<div data-test-block="true">
  <fieldset class="Fieldset_fieldset__gwnkr">
    <legend id="legendId-:r9:"
            class="Fieldset_legend__8odl-   Fieldset_legendNoMarginBottom__UO3iu"
    >
      Ledetekst
    </legend>
    <div class="Fieldset_description__k538X ">
      Kort hjelpetekst
    </div>
    <div class="Fieldset_contentContainer__fJcfT ">
      <div class>
        <label for="textFieldTextboxId-:rb:"
               class="TextField_label__K0isy  undefined"
        >
          <span class>
            Label
          </span>
        </label>
        <input id="textFieldTextboxId-:rb:"
               class="TextField_textbox__qbZlk    undefined"
               value
        >
        <div id="textFieldErrorId-:ra:"
             aria-live="assertive"
             aria-atomic="true"
        >
        </div>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Tester/Fieldset WithDisabled play-test 1`] = `
<div data-test-block="true">
  <fieldset class="Fieldset_fieldset__gwnkr"
            disabled
  >
    <legend id="legendId-:rp:"
            class="Fieldset_legend__8odl-"
    >
      Ledetekst
    </legend>
    <div class="Fieldset_contentContainer__fJcfT ">
      <div class>
        <label for="textFieldTextboxId-:rr:"
               class="TextField_label__K0isy  undefined"
        >
          <span class>
            Label
          </span>
        </label>
        <input id="textFieldTextboxId-:rr:"
               class="TextField_textbox__qbZlk    undefined"
               value
        >
        <div id="textFieldErrorId-:rq:"
             aria-live="assertive"
             aria-atomic="true"
        >
        </div>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Tester/Fieldset WithHelpSvgPath smoke-test 1`] = `
<div data-test-block="true">
  <fieldset class="Fieldset_fieldset__gwnkr">
    <legend id="legendId-:r11:"
            class="Fieldset_legend__8odl-   Fieldset_legendNoMarginBottom__UO3iu"
    >
      Ledetekst
    </legend>
    <button class="IconButton_button__mw9sG IconButton_button_extraSmall__3+QfY IconButton_button_outlined__aDbNH Fieldset_helpButton__Qd9mY "
            type="button"
            aria-describedby="legendId-:r11:"
            aria-expanded="false"
    >
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz IconButton_icon__1J4sN"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:r12:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:r12:">
          Hjelp
        </title>
        <path d="M13 14h-2v-4h2v4Zm0 4h-2v-2h2v2ZM1 21h22L12 2 1 21Z">
        </path>
      </svg>
    </button>
    <div class="Fieldset_contentContainer__fJcfT ">
      <div class>
        <label for="textFieldTextboxId-:r14:"
               class="TextField_label__K0isy  undefined"
        >
          <span class>
            Label
          </span>
        </label>
        <input id="textFieldTextboxId-:r14:"
               class="TextField_textbox__qbZlk    undefined"
               value
        >
        <div id="textFieldErrorId-:r13:"
             aria-live="assertive"
             aria-atomic="true"
        >
        </div>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Tester/Fieldset WithHelpText play-test 1`] = `
<div data-test-block="true">
  <fieldset class="Fieldset_fieldset__gwnkr">
    <legend id="legendId-:rs:"
            class="Fieldset_legend__8odl-   Fieldset_legendNoMarginBottom__UO3iu"
    >
      Ledetekst
    </legend>
    <button class="IconButton_button__mw9sG IconButton_button_extraSmall__3+QfY IconButton_button_outlined__aDbNH Fieldset_helpButton__Qd9mY "
            type="button"
            aria-describedby="legendId-:rs:"
            aria-expanded="false"
    >
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz IconButton_icon__1J4sN"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:rt:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:rt:">
          Hjelp
        </title>
        <path d="M10 19h3v3h-3v-3Zm2-17c5.35.22 7.68 5.62 4.5 9.67-.83 1-2.17 1.66-2.83 2.5C13 15 13 16 13 17h-3c0-1.67 0-3.08.67-4.08.66-1 2-1.59 2.83-2.25C15.92 8.43 15.32 5.26 12 5a3 3 0 0 0-3 3H6a6 6 0 0 1 6-6Z">
        </path>
      </svg>
    </button>
    <div class="Fieldset_contentContainer__fJcfT ">
      <div class>
        <label for="textFieldTextboxId-:rv:"
               class="TextField_label__K0isy  undefined"
        >
          <span class>
            Label
          </span>
        </label>
        <input id="textFieldTextboxId-:rv:"
               class="TextField_textbox__qbZlk    undefined"
               value
        >
        <div id="textFieldErrorId-:ru:"
             aria-live="assertive"
             aria-atomic="true"
        >
        </div>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Tester/Fieldset WithHideLegend play-test 1`] = `
<div data-test-block="true">
  <fieldset class="Fieldset_fieldset__gwnkr">
    <legend id="legendId-:ri:"
            class="Fieldset_legend__8odl-  Fieldset_srOnly__7qh39 Fieldset_legendNoMarginBottom__UO3iu"
    >
      Ledetekst
    </legend>
    <button class="IconButton_button__mw9sG IconButton_button_extraSmall__3+QfY IconButton_button_outlined__aDbNH Fieldset_helpButton__Qd9mY Fieldset_srOnly__7qh39"
            type="button"
            aria-describedby="legendId-:ri:"
            aria-expanded="false"
    >
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz IconButton_icon__1J4sN"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:rj:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:rj:">
          Hjelp
        </title>
        <path d="M10 19h3v3h-3v-3Zm2-17c5.35.22 7.68 5.62 4.5 9.67-.83 1-2.17 1.66-2.83 2.5C13 15 13 16 13 17h-3c0-1.67 0-3.08.67-4.08.66-1 2-1.59 2.83-2.25C15.92 8.43 15.32 5.26 12 5a3 3 0 0 0-3 3H6a6 6 0 0 1 6-6Z">
        </path>
      </svg>
    </button>
    <div class="Fieldset_description__k538X Fieldset_srOnly__7qh39">
      Kort hjelpetekst
    </div>
    <div class="Fieldset_contentContainer__fJcfT Fieldset_contentContainerNoMarginTop__YkL1P">
      <div class>
        <label for="textFieldTextboxId-:rl:"
               class="TextField_label__K0isy  undefined"
        >
          <span class>
            Label
          </span>
        </label>
        <input id="textFieldTextboxId-:rl:"
               class="TextField_textbox__qbZlk    undefined"
               value
        >
        <div id="textFieldErrorId-:rk:"
             aria-live="assertive"
             aria-atomic="true"
        >
        </div>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Tester/Fieldset WithRef play-test 1`] = `
<div data-test-block="true">
  <fieldset class="Fieldset_fieldset__gwnkr"
            id="dummyIdForwardedFromRef"
  >
    <legend id="legendId-:r0:"
            class="Fieldset_legend__8odl-"
    >
      Ledetekst
    </legend>
    <div class="Fieldset_contentContainer__fJcfT ">
      <div class>
        <label for="textFieldTextboxId-:r2:"
               class="TextField_label__K0isy  undefined"
        >
          <span class>
            Label
          </span>
        </label>
        <input id="textFieldTextboxId-:r2:"
               class="TextField_textbox__qbZlk    undefined"
               value
        >
        <div id="textFieldErrorId-:r1:"
             aria-live="assertive"
             aria-atomic="true"
        >
        </div>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Tester/Fieldset WithShowRequiredMark smoke-test 1`] = `
<div data-test-block="true">
  <fieldset class="Fieldset_fieldset__gwnkr">
    <legend id="legendId-:rc:"
            class="Fieldset_legend__8odl- Fieldset_legend_required__U4xWm"
    >
      Ledetekst
    </legend>
    <div class="Fieldset_contentContainer__fJcfT ">
      <div class>
        <label for="textFieldTextboxId-:re:"
               class="TextField_label__K0isy  undefined"
        >
          <span class>
            Label
          </span>
        </label>
        <input id="textFieldTextboxId-:re:"
               class="TextField_textbox__qbZlk    undefined"
               value
        >
        <div id="textFieldErrorId-:rd:"
             aria-live="assertive"
             aria-atomic="true"
        >
        </div>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Tester/Fieldset WithShowRequiredMarkAndLegend smoke-test 1`] = `
<div data-test-block="true">
  <fieldset class="Fieldset_fieldset__gwnkr">
    <legend id="legendId-:rf:"
            class="Fieldset_legend__8odl- Fieldset_legendWithMarkup_required__WcdUI"
    >
      <h1 class="Heading_heading__7K0um Heading_heading_level3__T5BVI  ">
        Dette er en Heading i legend
      </h1>
      <p class="Paragraph_paragraph__NPzKO Paragraph_paragraph_ingress__K1ALx  ">
        <em>
          Dette er en italic Paragraph med ingress variant i legend
        </em>
      </p>
    </legend>
    <div class="Fieldset_contentContainer__fJcfT ">
      <div class>
        <label for="textFieldTextboxId-:rh:"
               class="TextField_label__K0isy  undefined"
        >
          <span class>
            Label
          </span>
        </label>
        <input id="textFieldTextboxId-:rh:"
               class="TextField_textbox__qbZlk    undefined"
               value
        >
        <div id="textFieldErrorId-:rg:"
             aria-live="assertive"
             aria-atomic="true"
        >
        </div>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`Tester/Fieldset WithSpacing smoke-test 1`] = `
<div data-test-block="true">
  <fieldset class="Fieldset_fieldset__gwnkr Fieldset_fieldset_hasSpacing__966Le">
    <legend id="legendId-:rm:"
            class="Fieldset_legend__8odl-"
    >
      Ledetekst
    </legend>
    <div class="Fieldset_contentContainer__fJcfT ">
      <div class>
        <label for="textFieldTextboxId-:ro:"
               class="TextField_label__K0isy  undefined"
        >
          <span class>
            Label
          </span>
        </label>
        <input id="textFieldTextboxId-:ro:"
               class="TextField_textbox__qbZlk    undefined"
               value
        >
        <div id="textFieldErrorId-:rn:"
             aria-live="assertive"
             aria-atomic="true"
        >
        </div>
      </div>
    </div>
  </fieldset>
</div>
`;
